packages/backend/src/auth/middleware.ts

```diff
@@ -0,0 +1,28 @@
+import type { NextFunction, Request, Response } from 'express'
+import jwt from 'jsonwebtoken'
+import { config } from '../config'
+
+export interface RequestWithToken extends Request {
+  jwtPayload?: jwt.JwtPayload
+}
+
+export function authenticate(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void {
+  const { token } = req.cookies
+  if (token) {
+    const secretKey = config.auth.jwtSecretKey
+    const verification = jwt.verify(token, secretKey)
+    console.log('verification', verification)
+    if (typeof verification === 'string') {
+      next('Verification is string')
```

Comment:
I'm confused how this is used?

Reply:
The parameter you pass to the `next` function is considered an error. So it should throw an error and end up in our error handling middleware.

Reply:
I don't like this part of the code. I'll look at it again if we're able to make it better. I just don't understand why it can return a string and what that string means.

Reply:
I understand it now. If the signed payload with `jwt.sign` is string, then the result of the `jwt.verify` is gonna be a string. I'll update the condition/error message accordingly.

Reply:
I updated the error message.